feat(moderation): add guardrail checks and logging for moderation (#2624)
promptfoo/promptfoo: site/docs/configuration/expected-outputs/moderation.md

In general, we encourage the use of Meta's LlamaGuard as it substantially outperforms OpenAI's moderation API as well as GPT-4. [See benchmarks](https://github.com/meta-llama/PurpleLlama/blob/main/Llama-Guard2/MODEL_CARD.md#model-performance).

## Azure OpenAI content filters

> When using Azure AI model deployments, the moderation assertion utilizes Azure's built-in content filtering system, powered by Azure AI Content Safety. These content filters are configured during model deployment in Azure, allowing you to set appropriate severity thresholds for different categories of harmful content. This configuration ensures that the content filtering system enforces your specified moderation policies during model inference.

Reviewer: This sounds a little too much like AI, I think we can simplify

```suggestion
If you are using an Azure model, the moderation assertion uses the built-in "Azure AI Content Safety" system. These content filters are configured during model deployment in Azure.
```
